packaging/flavorgen/flavors/generators.go

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flavors
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha3"
+	cloudproviderSvc "sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/cloudprovider"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	kubeadmv1beta1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+)
+
+const (
+	clusterNameVar              = "${ CLUSTER_NAME }"
+	namespaceVar                = "${ NAMESPACE }"
+	defaultNumCPUs              = 2
+	defaultCloudProviderImage   = "gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.0.0"
+	kubernetesVersionVar        = "${ KUBERNETES_VERSION }"
+	workerMachineCountVar       = "${ WORKER_MACHINE_COUNT }"
+	controlPlaneMachineCountVar = "${ CONTROL_PLANE_MACHINE_COUNT }"
+	folderVar                   = "${ VSPHERE_FOLDER_VAR }"
+	machineDeploymentNameSuffix = "-md-0"
+	resourcePoolVar             = "${ VSPHERE_RESOURCE_POOL }"
+	sshAuthorizedKeysVar        = "${ SSH_AUTHORIZED_KEY }"
+	vSphereDataCenterVar        = "${ VSPHERE_DATACENTER }"
+	vSphereDatastoreVar         = "${ VSPHERE_DATASTORE }"
+	vSphereNetworkVar           = "${ VSPHERE_NETWORK }"
+	vSphereServerVar            = "${ VSPHERE_SERVER }"
+	vSphereTemplateVar          = "${ VSPHERE_TEMPLATE }"
+	defaultHAProxyTemplate      = "capv-haproxy-v0.5.3-77-g224e0ef6"
```

Comment:
this shouldn't be hardcoded. this needs to be provided by the user. consumers can have different names for this OVA